app/mm-redux/utils/post_utils.ts

```diff
@@ -62,11 +62,18 @@ export function canDeletePost(state: GlobalState, config: any, license: any, tea
     const isOwner = isPostOwner(userId, post);
 
     if (hasNewPermissions(state)) {
-        const canDelete = haveIChannelPermission(state, {team: teamId, channel: channelId, permission: Permissions.DELETE_POST});
-        if (!isOwner) {
-            return canDelete && haveIChannelPermission(state, {team: teamId, channel: channelId, permission: Permissions.DELETE_OTHERS_POSTS});
+        let permissions = [];
+        if (isOwner) {
+            permissions = [Permissions.DELETE_POST];
+        } else {
+            const {serverVersion} = state.entities.general;
+
+            // prior to v5.27, the server used to require delete_own_posts and
+            // delete_others_posts permissions to be able to delete a post by a
+            // different author.
+            permissions = isMinimumServerVersion(serverVersion, 5, 27) ? [Permissions.DELETE_OTHERS_POSTS] : [Permissions.DELETE_POST, Permissions.DELETE_OTHERS_POSTS];
         }
-        return canDelete;
+        return permissions.every((permission) => haveIChannelPermission(state, {team: teamId, channel: channelId, permission, default: false}));
```

Comment:
I kind of don't like this..

this is running a "selector" which is not really functioning as a selector cause the last argument always changes and now it must run potentially 3 times (one per each permission) every time anything in the redux state changes.

Reply:
Isn't this the same as it's being done in the function below? https://github.com/mattermost/mattermost-mobile/pull/4660/files#diff-e45c412d1f7a1830bb37bb0a57c5eae1R108

Reply:
It could be, I’m just exposing my concern

Reply:
Every time `canDeletePost` is invoked, `haveIChannelPermission` needs to be invoked once for each distinct permission being checked. Is that happening any more times because of the way this is written?

Reply:
Well we could define an object as a constant outside the function and set its properties with the channelID and teamID values, that way is always the same object and does. Lt need to re-compute as it will be memoized, right?

Maybe a selector factory that receives team and channel id’s as args instead of an object?

I do think there are alternatives

Reply:
It sounds like you're proposing a refactor to the `haveIChannelPermission` selector, which is outside of the scope of this change. However, I'm interested to see if what you're proposing can provide an optimization to the permissions selectors—they're used extensively throughout the browser, desktop, and mobile clients. Likely it would involve a refactor to `haveIChannelPermission`, `haveITeamPermission`, and `haveISystemPermission` (and several of their helpers).

Reply:
Ok I get is out of scope and I'm fine with it.

Maybe as we look into v2 we can come up with a better way to do this to avoid something like this that can have a small impact but as it is being used all over the codebase it all sums up.

Reply:
Yes, you're definitely correct that those permission selectors are not as efficient as possible. Doing a `console.count` inside of `haveIChannelPermission` for a given permission ID, for example, shows that the function's body is invoked more times than once per channel per permission ID.

Reply:
I wonder if we should create a ticket to improve on it regardless of v2. v2 is going to be ready sometime Q1/Q2 2021, so until then this is still relevant as well as it is being used on the webapp. IMHO we should give this another look